sycl/doc/extensions/Matrix/dpcpp-joint-matrix.asciidoc

```diff
@@ -0,0 +1,304 @@
+//Assuming memB has already been VNNIed
+q.parallel_for(nd_range<2>(G, L), [=](nd_item<2> item)                            
+  [[sycl::reqd_sub_group_size(SG_SIZE)]] {
+   const auto global_idx = item.get_global_id(0);
+   const auto global_idy = item.get_global_id(1);
+   const auto sg_startx = global_idx - item.get_local_id(0);
+   const auto sg_starty = global_idy - item.get_local_id(1);
+   sub_group sg = item.get_sub_group();
+   joint_matrix<sub_group, int8_t, tM, tK> tA(sg);
+   // For B, since current implementation does not support non packed layout,
+   // users need to specify the updated VNNI sizes along with the packed_b layout
+   joint_matrix<sub_group, int8_t, tK, tN, packed_b> tB(sg);
+   joint_matrix<sub_group, int32_t, tM, tN> tC(sg);
+   joint_matrix_load(sg, tC, memC + sg_startx * tM * N + sg_starty, N, matrix_layout::row_major);
+   for (int k = 0; k < K; k += tk) {
+     joint_matrix_load(sg, tA, memA + sg_startx * tM * K + k, K, matrix_layout::row_major);//collective
+     joint_matrix_load(sg, tB, memB + k * N + sg_starty, N, matrix_layout::packed_b);//VNNI
+     tC = joint_matrix_mad(sg, tA, tB, tC);
+   }
+   joint_matrix_store(sg, tC, memC + sg_startx * tM * N + sg_starty, N, matrix_layout::row_major);
+}).wait();
+  
+```
+## Implementation Status
+For oneAPI release 3, an AOT implementation is available on the CPU device to targets AMX hardware. we are using AMX tile intrinsics to implement the matrix load and store operations. Since we are currently emitting AMX intrinsics directly, this only enables AOT compilation. 
+
+Currently, this is the compilation command line needed to invoke AMX unit of Sapphire Rapids CPU:
+
+```c++
+clang++ -fsycl -march=sapphirerapids fsycl-targets="spir64_x86_64-uknown-linux-sycldevice"  -O2 matmul-int8.cpp -o matmul-int8
+```
+
+Please refer to the section "Future Implementation Work" that talks about the future unified SPIR-V path that will enable JIT compilation.
+
+// We used the https://software.intel.com/sites/landingpage/IntrinsicsGuide/#techs=AMX[`_tile_`-prefixed intrinsics] defined in `immintrin.h`.
+
+
+### Current Implementation Restrictions
+This section summarizes the specific features that this implementation supports. In future versions of this API and implementation, the expectation is to provide a query interface to guide the usage of this API. 
+
+#### Type, Sizes, and Layouts
+The types supported by this AMX implementation are restricted to the types that AMX hardware support. Although the AMX hardware supports 2d tiles with a maximum size of 16x64 bytes, this current implementation can handle any size. If the matrix size is bigger than 1024 bytes, it will be stored in memory rather than mapped to a 2d tile. Performance penalty may occur in this case. In order to get the best performance with this implementation, matrix sizes should be no larger than 16x64 bytes and A and B matrices should be already packed (put in VNNI format).
```

Comment:
> In order to get the best performance with this implementation, matrix sizes should be no larger than 16x64 bytes and A and B matrices should be already packed (put in VNNI format).

This seems to contradict the earlier statements that only matrix B must be packed onto VNNI format, and that matrix A must be in row-major format.

Reply:
I removed A from the text.